.github/workflows/publish.yaml

```diff
@@ -0,0 +1,26 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*.*.*'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    env:
+      CONTAINER_ENGINE: podman
+      CHECKUP_IMAGE_TAG: ${{github.ref_name}}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Login to quay.io
+        run:
+          ${CONTAINER_ENGINE} login -u ${{ secrets.QUAY_USER }} -p ${{ secrets.QUAY_TOKEN }} quay.io
+      - name: Build and push image
+        run: make push
```

Comment:
What about build-test-lint?

Reply:
At this stage, the project was already linted, unit tested and built.
I don't see a reason to do it again here.
The project is built again here only because workflows are independent and requires more work to share data between them - which could be invested in the future.